[website] Open 4 new roles (#27123)
mui-org/material-ui: docs/src/pages/company/careers/careers.md

The open-source project started back in 2014 to unify React and Material Design. Today, Material-UI has grown to become one of the world's most popular React UI libraries, backed by a vibrant community of more than 2M developers in over 180 countries.

> The company is bootstrapped (so far). It was incorporated in mid-2019 and yet growing fast (x2-3 YoY). We doubled the team in 2020 and 2021.

Reviewer: It's still 2021 :)

### Engineering

#### [Full-stack Engineer](/company/full-stack-engineer/)

> You will initiate the development of a bold new product vertical. We are looking for an experienced, and ambitious full-stack engineer that is ready to work in an entrepreneurial environment.

Reviewer: ```suggestion
You will initiate the development of a bold new product vertical. We are looking for an experienced and ambitious full-stack engineer that is ready to work in an entrepreneurial environment.
```

You are a [manager of one](https://signalvnoise.com/posts/1430-hire-managers-of-one), you are curious, enjoy taking risks, and learning.

#### [React Engineer](/company/react-engineer/)

> You will support the advanced components team, build new ambitious enterprise features, work on strategical problems, and grow the open-source adoption of the free tier.

Reviewer: ```suggestion
You will support the advanced components team, build new ambitious enterprise features, work on strategic problems, and help grow the adoption of the free open-source tier.
```

### Product

#### [Technical Product Manager](/company/product-manager/)

> You will define and maintain the product roadmap for the advanced components, identify opportunities, define specs, work with engineers to execute on the features.

Reviewer: ```suggestion
You will define and maintain the product roadmap for the advanced components, identify opportunities, define specs, and work with engineers to execute on the features.
```

> An experience as an engineer is essential for this role.

Reviewer: ```suggestion
Experience as an engineer is essential for this role.
```

#### [Developer Advocate](/company/developer-advocate/)

> You will educate on the latest features, craft high-quality examples and demos, engage with the community, write documentation, advocate for creating faster and more appealing UIs, help promote/market the advanced components.

Reviewer: ```suggestion
You will educate users on the latest features, craft high-quality examples and demos, engage with the community, write documentation, advocate for creating faster and more appealing UIs, and help to promote/market the advanced components.
```
